docs/index.rst

```diff
@@ -1,8 +1,76 @@
 ``pluggy``
 ==========
 
-The ``pytest`` plugin system
-****************************
+The five minute elevator pitch
+******************************
+
+``pluggy`` gives `users` the possibility to extend or modify the behaviour of
+a program like e.g. `pytest`_, `tox`_ or `devpi`_ by installing a `plugin` for
+that program. The plugin code will run as part of normal program execution
+changing or enhancing certain aspects of it.
+
+There are three roles in which people might interact with ``pluggy``.
+
+Publisher
++++++++++
+
+The `publisher` wants to enable an arbitrary number of `subscribers` to extend
+or modify some aspects of their programs execution. They write hook specifications,
+register them with a :py:class:`pluggy.PluginManager` and instantiate a marker
+object for the `subscriber` - a :py:class:`~pluggy.HookimplMarker` - used
+to `decorate <https://www.python.org/dev/peps/pep-0318/#current-syntax>` hook
+implementation functions. This marker is usually called `hookimpl` and
+importable from the root package of the `publisher` project. They then add
+calls to their specified hooks in appropriate phases of program execution.
+``pluggy`` will then do all the magic.
+
+**FIXME not sure what is the right advice here. Do I need to vendor in pluggy as a publisher to avoid versioning conflicts? Should I just depend on pluggy without version and hope all is good?**
```

Comment:
I'd say latter is what we recommend; so far I don't think we've had any breaking changes particularly because `pytest` and friends are all dependants.

Reply:
o.k. that might be the sanest approach then :)

Reply:
pytest does `'pluggy>=0.5,<0.7'` and devpi `pluggy>=0.3.0,<1.0` :scream:

Reply:
In this regard I think we can all agree on pinning `>=0.x,<1.0`; we definitely don't want to introduce changes that break pytest, devpi or tox even in pre-1.0. 

And after all we are all the developers of the same projects, so we can all agree to postpone any breaking changes until 1.0.

Reply:
i wouldn't want to put that limit on per se - but we certainly can coordinate fallout

technically we should probably do a 1.0 already since pluggy is in productive use, and ever since https://0ver.org/ its painfully oblivious that that is just wrong ^^

Reply:
0ver.org ... :laughing: reminds me of flask. We're using that in production since 2011 and it is at 0.12.2 atm.

Reply:
> technically we should probably do a 1.0 already since pluggy is in productive use,

Agree, let's make the next one `1.0` already then, or are there any significant features to be added before that?